site/content/features/textblocks.md

```diff
@@ -0,0 +1,444 @@
+
+## Whitespace
+
+Perhaps surprisingly, the biggest complexity with text blocks comes from the handling of white space. There are three distinct mechanisms at play: normalizing line endings, stripping trailing whitespace, and stripping common indent. We will examine all three in the following sections.
+
+Note that whitespace is handled *before* escapes (including line continuations) are processed. 
+
+### Normalizing Line Endings
+
+Line endings in text blocks (other than the one after the opening `"""` and after continuation escapes) are turned into `\n`, even if the source file uses DOS-style line endings (`\r\n`). That is sensible—you wouldn't want code like the following depend on whether the source file was edited on Windows or Linux:
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+public class Sandbox {
+    public static void main(String[] args) {
+        String greeting = """
+Hello
+World""";
+        System.out.println(greeting.length());
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+### Removing Trailing White Space
+
+Trailing whitespace is removed from each line. That way, there are no surprises if an invisible space is added accidentally. That is also sensible, and consistent with ignoring whitespace after the opening `"""`.
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+public class Sandbox {
+    public static void main(String[] args) {
+        String greeting = """
+Hello   
+World   """;
+        System.out.println("Trailing whitespace removed: ");
+        System.out.println("|" + greeting + "|");
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+To preserve trailing whitespace, use the `\s` escape. In this prompt, the trailing spaces are not removed:
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+import java.util.Scanner;
+
+public class Sandbox {
+    public static void main(String[] args) {
+        String prompt = """
+Hello, my name is Hal.
+Please enter your name:\s""";
+        System.out.print(prompt);
+        String simulatedInput = "Fred";
+        System.out.println(simulatedInput);
+        System.out.printf("Hello, %s, nice to meet you!%n", simulatedInput);
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+Any spaces before the `\s` are preserved as well:
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+import java.util.stream.Stream;
+
+public class Sandbox {
+    public static void main(String[] args) {
+        String snowmanWithColoredBackground = """
+<pre style="background: skyblue;">
+        \s
+   (")  \s
+  ( : ) \s
+ (     )\s
+---------
+</pre>""";
+        Stream.of(snowmanWithColoredBackground.split("\n")).forEach(line -> System.out.println("|" + line + "|"));
```

Comment:
With `String.lines()` this can be written more compact:

    snowmanWithColoredBackground.lines().forEach(line -> System.out.println("|" + line + "|"));

Reply:
Thank you, much nicer! I changed that.